### ISSUE TITLE
Consistent Numpy underflow error on some machines

### ISSUE THREAD

**Issue by willdumm** [matsengrp/gctree#67]
Related to #6

On some machines (my MacBook Pro, or FH Gizmos/cluster) running `gctree infer` on the sample data results in a consistent numpy underflow error, while on other machines (quokka) it does not.

I suspect this could be an issue with dynamically linked libraries implementing numpy's floating point arithmetic. Indeed, when running on the Fred Hutch cluster, I'm using _exactly the same_ Conda installation and environment as I use on quokka. On both platforms, float64 has the same precision. What else could be different?

Numpy version on all machines tested: 1.21.3
scipy: 1.7.1

output:
```
number of trees with integer branch lengths: 203
203 trees exhibit unobserved unifurcation from root. Adding psuedocounts to these roots
floating point error in MLE: underflow encountered in exp. Attempt 1 of 10. Rerunning with new random start.
floating point error in MLE: underflow encountered in exp. Attempt 2 of 10. Rerunning with new random start.
floating point error in MLE: underflow encountered in exp. Attempt 3 of 10. Rerunning with new random start.
floating point error in MLE: underflow encountered in exp. Attempt 4 of 10. Rerunning with new random start.
floating point error in MLE: underflow encountered in exp. Attempt 5 of 10. Rerunning with new random start.
floating point error in MLE: underflow encountered in exp. Attempt 6 of 10. Rerunning with new random start.
floating point error in MLE: underflow encountered in exp. Attempt 7 of 10. Rerunning with new random start.
floating point error in MLE: underflow encountered in exp. Attempt 8 of 10. Rerunning with new random start.
floating point error in MLE: underflow encountered in exp. Attempt 9 of 10. Rerunning with new random start.
Traceback (most recent call last):
  File "/Users/wdumm/opt/miniconda3/envs/gctree/bin/gctree", line 8, in <module>
    sys.exit(main())
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/cli.py", line 749, in main
    args.func(args)
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/cli.py", line 220, in infer
    p, q = parsimony_forest.mle(marginal=True)
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/branching_processes.py", line 864, in mle
    return _mle_helper(self.ll, **kwargs)
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/branching_processes.py", line 886, in _mle_helper
    grad_check = check_grad(lambda x: f(x)[0], lambda x: f(x)[1], x_0)
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/scipy/optimize/optimize.py", line 974, in check_grad
    return sqrt(sum((grad(x0, *args) -
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/branching_processes.py", line 886, in <lambda>
    grad_check = check_grad(lambda x: f(x)[0], lambda x: f(x)[1], x_0)
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/branching_processes.py", line 884, in f
    return tuple(-x for x in ll(*x, **kwargs))
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/branching_processes.py", line 839, in ll
    CollapsedTree._build_ll_genotype_cache(self._c_max, self._m_max, p, q)
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/branching_processes.py", line 289, in _build_ll_genotype_cache
    CollapsedTree._ll_genotype(c, m, p, q)
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/gctree/branching_processes.py", line 275, in _ll_genotype
    logf_result = logsumexp(logg_array)
  File "/Users/wdumm/opt/miniconda3/envs/gctree/lib/python3.10/site-packages/scipy/special/_logsumexp.py", line 110, in logsumexp
    tmp = np.exp(a - a_max)
FloatingPointError: underflow encountered in exp
```

**Comment by willdumm** [comment]
I've isolated this a bit. A minimal example to reproduce the error is:
```
import gctree
gctree.CollapsedTree._ll_genotype(0, 3, 0.5, 0.5)
```
This will raise the FloatingPointError that's causing the problem, on some machines.

The error is coming from line 275, in a call to spicy.special.logsumexp. It can be reproduced with:
```
from scipy.special import logsumexp
import numpy as np
logsumexp([-3, -np.inf, -np.inf])
```

I also noticed while debugging this that although the error message suggests that MLE will be tried again with a new random start, the initial parameters are actually fixed. It seems like this should be fixed, by either revising the error message (and is it necessary to try ten times with the same starting parameters) or actually allowing branching_processes._mle_helper to pick random parameters.
